storybook/storybook-react/src/Button/Button.stories.tsx

```diff
@@ -25,16 +41,93 @@ export const Primary: Story = {
   },
 }
 
+export const PrimaryWithIcon: Story = {
+  args: {
+    children: ['Primary', <Icon key="icon" svg={ShareIcon} size="level-5" />],
+  },
+}
+
 export const Secondary: Story = {
   args: {
     children: 'Secondary',
     variant: 'secondary',
   },
 }
 
+export const SecondaryWithIcon: Story = {
+  args: {
+    variant: 'secondary',
+    children: ['Secondary', <Icon key="icon" svg={ShareIcon} size="level-5" />],
+  },
+}
+
+export const SecondaryWithIconAndBadge: Story = {
+  args: {
+    variant: 'secondary',
+    children: [
+      'Secondary',
+      <Badge key="badge" label={99} color="dark-blue" />,
+      <Icon key="icon" svg={CloseIcon} size="level-5" />,
+    ],
+  },
+}
```

Comment:
I don't think you have to show you can add other components to Button 4 times, PrimaryWithIcon is enough imo

Reply:
This was left in here on purpose to test the theory that we don't need a filter button

Reply:
Changed it and moved busy/disabled to controls. I kept secondary with a badge as an example for the filter button discussion. Is that ok?

Reply:
I'm not sure we need to show a filter button with a Badge, that design doesn't seem to be very fleshed out. In most examples, a filter button is a Button with a Close Icon. That's all I would show here. Maybe we should call the story 'Button with an icon', so we don't imply that only the Primary Button can have an Icon. I'd move it to after the Tertiary Button story, so that distinction is more clear